packages/core/src/dialects/cockroachdb/query-interface.ts

```diff
@@ -0,0 +1,45 @@
+import type { QueryRawOptions, RemoveConstraintOptions } from '../../index';
+import type { TableNameOrModel } from '../abstract/query-generator-typescript';
+import { PostgresQueryInterface } from '../postgres/query-interface';
+
+export class CockroachDbQueryInterface extends PostgresQueryInterface {
+  async dropSchema(schema: string, options: QueryRawOptions) {
+    if (schema === 'crdb_internal') {
+      throw new Error('Cannot remove crdb_internal schema in Cockroachdb');
+    }
+
+    await super.dropSchema(schema, options);
+  }
+
+  // CockroachDB support dropping constraints like Postgres unless the constraint is being referenced by a partial index predicate.
+  // In such cases only DROP index is to be used https://github.com/cockroachdb/cockroach/issues/97813
+  async removeConstraint(
+    tableName: TableNameOrModel,
+    constraintName: string,
+    options: RemoveConstraintOptions,
+  ): Promise<any> {
+    try {
+      await super.removeConstraint(tableName, constraintName, options);
+    } catch (error: any) {
+      if (error.message.includes('use DROP INDEX CASCADE instead')) {
```

Comment:
I don't think the code should try to correct for this. It should alert the user, and the user should then use the `cascade` option themselves, that option is already supported in `RemoveConstraintQueryOptions`